### PR TITLE
update split feature file and step definition, adding constructor

### PR DESCRIPTION
**Pull request by MaeveMai**

### CODE REVIEW

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/CertificationModule.cs [hunk 1]
```diff
@@ -10,14 +10,23 @@ namespace MVPMarsOnboarding.Pages
 {
     public class CertificationModule
     {
-        public string GetNewCertificateName(IWebDriver driver)
+        public IWebDriver driver;
+        CertificationModule CertificationModuleObj;
+        ProfilePage profilePageObj;
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/CertificationModule.cs]: I would do this in steps remember to keep the page logic away from other pages

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/CertificationModule.cs [hunk 2]
```diff
@@ -10,14 +10,23 @@ namespace MVPMarsOnboarding.Pages
 {
     public class CertificationModule
     {
-        public string GetNewCertificateName(IWebDriver driver)
+        public IWebDriver driver;
+        CertificationModule CertificationModuleObj;
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/CertificationModule.cs]: do you need this?

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/CertificationModule.cs [hunk 3]
```diff
@@ -10,14 +10,23 @@ namespace MVPMarsOnboarding.Pages
...
+
+        public string GetNewCertificateName()
         {
             try
             {
                 //refresh the page
                 driver.Navigate().Refresh();
-                ProfilePage profilePageObj = new ProfilePage();
-                profilePageObj.GoToCertificationModule(driver);
+                profilePageObj = new ProfilePage(driver);
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/CertificationModule.cs]: should move to the profile page, also this profilePageObj = new ProfilePage(driver); should be done in the constructor,  

 public CertificationModule(IWebDriver driver)
        {
            this.driver = driver;

           // as an example but this should be present here
            this.profilePageObj = new ProfilePage(driver);
        }

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/EducationModule.cs [hunk 4]
```diff
@@ -10,14 +10,22 @@ namespace MVPMarsOnboarding.Pages
...
+
+        public string GetNewUniversityName()
         {
             try
             {
                 //refresh the page
                 driver.Navigate().Refresh();
-                ProfilePage profilePageObj = new ProfilePage();
-                profilePageObj.GoToEducationModule(driver);
+                ProfilePage profilePageObj = new ProfilePage(driver);
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/EducationModule.cs]: same here

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/LanguageModule.cs [hunk 5]
```diff
@@ -11,15 +11,23 @@ namespace MVPMarsOnboarding.Pages
 {
     public class LanguageModule:CommonDriver
     {
+        public IWebDriver driver;
+        LanguageModule languageModuleObj;
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/LanguageModule.cs]: not needed i believe

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/LanguageModule.cs [hunk 6]
```diff
@@ -11,15 +11,23 @@ namespace MVPMarsOnboarding.Pages
...
+        public string GetNewLanguage()
+        {
+            
             try
             {
                 //refresh the page
                 driver.Navigate().Refresh();
-                ProfilePage profilePageObj = new ProfilePage();
-                profilePageObj.GoToLanguageModule(driver);
+                ProfilePage profilePageObj = new ProfilePage(driver);
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/LanguageModule.cs]: is this the profile page

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/ProfilePage.cs [hunk 7]
```diff
@@ -11,28 +11,35 @@ namespace MVPMarsOnboarding.Pages
     public class ProfilePage
 
     {
-        public string GetHiUser(IWebDriver driver)
+        public IWebDriver driver;
+
+        public ProfilePage(IWebDriver driver)
+        {
+            this.driver = driver;
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/Pages/ProfilePage.cs]: this looks good to me the constructor

File: MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/CertificateModuleStepDefinition.cs [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+    public class CertificateModuleStepDefinition : CommonDriver
+    {
+
+        ProfilePage profilePageObj;
+        LoginPage loginPageObj;
+        CertificationModule certificationModuleObj;
+
+        public CertificateModuleStepDefinition()
+        {
+            driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/CertificateModuleStepDefinition.cs:18]: the driver you can initial once

File: MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/EducationModuleStepDefinition.cs [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+    [Binding]
+    public class EducationModuleStepDefinition:CommonDriver
+    {
+        ProfilePage profilePageObj;
+        LoginPage loginPageObj;
+        EducationModule educationModuleObj;
+
+        public EducationModuleStepDefinition()
+        {
+            driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/EducationModuleStepDefinition.cs:17]: initialise once in the Common driver

File: MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/LanguageModuleStepDefinition.cs [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+    [Binding]
+    public class LanguageModuleStepDefinition : CommonDriver
+    {
+        ProfilePage profilePageObj;
+        LoginPage loginPageObj;
+        LanguageModule LanguageModuleObj;
+
+        public LanguageModuleStepDefinition()
+        {
+            driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/LanguageModuleStepDefinition.cs:19]: initialise once in the Common driver

File: MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/LoginStepDefinition.cs [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    [Binding]
+    public class LoginStepDefinition:CommonDriver
+    {
+
+        ProfilePage profilePageObj;
+        LoginPage loginPageObj;
+
+        public LoginStepDefinition()
+        {
+            driver = new ChromeDriver();
```
Comment (amanbansalnz) [inline MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/LoginStepDefinition.cs:19]: initialise once in the Common driver

File: MVPMarsOnboarding/MVPMarsOnboarding/Pages/LanguageModule.cs (hunk 6 above)
Comment (MaeveMai) [reply MVPMarsOnboarding/MVPMarsOnboarding/Pages/LanguageModule.cs]: yes, This is I'm trying to refresh the profile page to see the change of language, hope I'm doing right

File: MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/EducationModuleStepDefinition.cs (hunk 9 above)
Comment (amanbansalnz) [reply MVPMarsOnboarding/MVPMarsOnboarding/StepDefinitions/EducationModuleStepDefinition.cs:17]: just not you dont need this driver = new ChromeDriver(); as you have done this in common driver :)

**Review by amanbansalnz — APPROVED** [review]

**Pull request merged**